paramiko/agent.py

```diff
@@ -412,7 +423,7 @@ def sign_ssh_data(self, data):
         msg.add_byte(cSSH2_AGENTC_SIGN_REQUEST)
         msg.add_string(self.blob)
         msg.add_string(data)
-        msg.add_int(0)
+        msg.add_int(SHA_MAP[self.name])
```

Comment:
https://tools.ietf.org/id/draft-miller-ssh-agent-01.html#sigflagnum
choose sha strategy based on name; sha1 = 0, sha256 = 2, sha512 = 4, bit 1 is reserved